nbs/15_gsa_gls/07-compile_gwas_snps_and_twas_genes.ipynb

```diff
@@ -0,0 +1,2355 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spredixcan_genes_models.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ef1c76d7-485b-4e8a-932e-fd01382d5520",
+   "metadata": {},
+   "source": [
+    "### Add gene variance per tissue"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "530f7941-6ad7-4a29-b9e4-a006949282e0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def _get_gene_variances(gene_row):\n",
+    "    gene_id = gene_row.name\n",
+    "    gene_tissues = gene_row[\"tissue\"]\n",
+    "\n",
+    "    tissue_variances = {}\n",
+    "    gene_obj = spredixcan_gene_obj[gene_id]\n",
+    "\n",
+    "    for tissue in gene_tissues:\n",
+    "        tissue_var = gene_obj.get_pred_expression_variance(\n",
+    "            tissue=tissue,\n",
+    "            reference_panel=REFERENCE_PANEL,\n",
+    "            model_type=EQTL_MODEL,\n",
+    "            snps_subset=gwas_variants_ids_set,\n",
+    "        )\n",
+    "\n",
+    "        if tissue_var is not None:\n",
+    "            tissue_variances[tissue] = tissue_var\n",
+    "\n",
+    "    return tissue_variances"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "01eb9e6a-4505-44e6-9e6e-9788e257f319",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "_tmp = spredixcan_genes_models.loc[\"ENSG00000000419\"]\n",
+    "_get_gene_variances(_tmp)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "447061ec-4847-4911-a1d6-8b86d15e4aac",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spredixcan_genes_tissues_variance = spredixcan_genes_models.apply(\n",
+    "    _get_gene_variances, axis=1\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "816cbdd2-d667-4eaa-bbf7-335550854ee8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spredixcan_genes_tissues_variance"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "68c65eb6-15c0-494d-992f-93bc84268a59",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# # testing\n",
+    "# _gene_id = \"ENSG00000188976\"\n",
+    "# x = spredixcan_genes_tissues_variance.loc[_gene_id]\n",
+    "# # expected value obtained by sum of PCA eigenvalues on this gene's predicted expression\n",
+    "# assert np.sum(list(x.values())) == pytest.approx(1.2326202607409493)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1996453a-fdfc-44e9-bbdf-b83e3b3b1aee",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# testing\n",
+    "spredixcan_genes_tissues_variance.loc[\"ENSG00000000419\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "62572b31-7e49-4ed0-ac1b-42552f3e264f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# FIXME: maybe add more tests, these differt from GTEX V8\n",
```

Comment:
Do you want to FIXME in this PR or leave it for later? :sweat_smile:

Reply:
Haha, no, these comments are old. Everything was already tested :)